### ISSUE TITLE
Add option to make @Module an error on companion objects

### ISSUE THREAD

**Issue by gabrielittner** [google/dagger#1742]
In the release notes you say:

>  For backwards compatibility, we still allow `@Module` on the companion object and @JvmStatic on the provides methods. However, `@Module` on the companion object is now a no-op and all of its attributes (e.g. "includes") will be ignored. In future releases, we will make it an error to use `@Module` on a companion object.

Could you add a processor option that already enables this behavior? When we updated to the latest release we already removed `@Module` from all companion objects. The option would help us ensure that no one in the team adds some again out of habit.

**Comment by Chang-Eric** [comment]
Yup, this is something we've already planned to do as part of changing the behavior.
